packages/powershell/oh-my-posh/oh-my-posh.psm1

```diff
@@ -17,7 +17,28 @@ function Get-PoshCommand {
 # Set the right binary to executable before doing anything else
 if ($PSVersionTable.PSEdition -eq "Core" -and !$IsWindows) {
     $executable = Get-PoshCommand
-    Invoke-Expression -Command "chmod +x $executable"
+    if (Test-Path $executable) {
+        # Check the permissions on the file
+        $permissions = ((ls -l $executable) -split ' ')[0]
+        if ((id -u) -eq 0) {
+            # Running as root, give global permissions if needed
+            $hasWrite = $permissions[2] -eq 'w'
+            $hasExecutable = $permissions[3] -eq 'x'
+            if ($hasWrite -and -not $hasExecutable) {
+                Invoke-Expression -Command "chmod g+x $executable"
+            }
+        }
+        else {
```

Comment:
rather than `if..else...` return early in the previous statements. Unless there's an actual use-case try to avoid if else as much a you can.

Reply:
This code is not in a function, so you can't really return.  If it even worked, it would skip all of the other function declarations and the Set-Aliases later in the script.

Reply:
You're right, didn't take that into account, reviews using the mobile app aren't all that... In that case, let's make it an internal function and call that.